src/wurf/dependency.py

```diff
@@ -79,34 +91,52 @@ def __init__(self, **kwargs):
         #    in external formats such as files
         # 2. Use the Python hash() function to generate a integer hash for 
         #    run-time usage (i.e. within the same interpreter session).
-        self.hash = None
+        #super(Dependency, self).__setattr__('hash', None)
+        self.read_only['hash'] = None
+        
+        # Resolver attributes (modifiable)
+        object.__setattr__(self, 'read_write', dict())
+        
         
     def __getattr__(self, option):
```

Comment:
Yes, the names in this class are not really good.. I will go with `attribute` since `property` is also a python built-in